File: src/main/java/de/tectoast/emolga/commands/flo/UpdateRemindersCommand.java
package de.tectoast.emolga.commands.flo;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.sql.DBManagers;
import net.dv8tion.jda.api.JDA;

import java.util.concurrent.Executors;

import static de.tectoast.emolga.utils.Constants.CALENDAR_MSGID;
import static de.tectoast.emolga.utils.Constants.CALENDAR_TCID;

public class UpdateRemindersCommand extends Command {

    public UpdateRemindersCommand() {
        super("updatereminders", "Updated die Reminder lol", CommandCategory.Flo);
    }

    @Override
    public void process(GuildCommandEvent e) {
        calendarService.shutdownNow();
        calendarService = Executors.newScheduledThreadPool(10);
        JDA jda = e.getJDA();
        jda.getTextChannelById(CALENDAR_TCID).editMessageById(CALENDAR_MSGID, buildCalendar()).queue();
        DBManagers.CALENDAR.getAllEntries().forEach(Command::scheduleCalendarEntry);
        e.done();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/flo/UpdateRemindersCommand.kt
package de.tectoast.emolga.commands.flo

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.utils.Constants.CALENDAR_MSGID
import de.tectoast.emolga.utils.Constants.CALENDAR_TCID
import de.tectoast.emolga.utils.sql.managers.CalendarManager

class UpdateRemindersCommand : Command("updatereminders", "Updated die Reminder lol", CommandCategory.Flo) {
    override fun process(e: GuildCommandEvent) {
        calendarService.shutdownNow()
        newCalendarService()
        e.jda.getTextChannelById(CALENDAR_TCID)!!.editMessageById(CALENDAR_MSGID, buildCalendar()).queue()
        CalendarManager.allEntries.forEach { scheduleCalendarEntry(it) }
        e.done()
    }

    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }
}